// framework7-vue/esm/framework7-vue-bundle.js
/**
 * Framework7 Vue 6.0.17
 * Build full featured iOS & Android apps using Framework7 & Vue
 * https://framework7.io/vue/
 *
 * Copyright 2014-2021 Vladimir Kharlampidi
 *
 * Released under the MIT License
 *
 * Released on: April 30, 2021
 */
import Framework7Vue from './shared/plugin';
import { f7, f7ready, theme } from './shared/f7';
import { useStore } from './shared/use-store';

import f7AccordionContent from './components/accordion-content';
import f7AccordionItem from './components/accordion-item';
import f7AccordionToggle from './components/accordion-toggle';
import f7Accordion from './components/accordion';
import f7ActionsButton from './components/actions-button';
import f7ActionsGroup from './components/actions-group';
import f7ActionsLabel from './components/actions-label';
import f7Actions from './components/actions';
import f7App from './components/app';
import f7Appbar from './components/appbar';
import f7AreaChart from './components/area-chart';
import f7Badge from './components/badge';
import f7BlockFooter from './components/block-footer';
import f7BlockHeader from './components/block-header';
import f7BlockTitle from './components/block-title';
import f7Block from './components/block';
import f7Button from './components/button';
import f7CardContent from './components/card-content';
import f7CardFooter from './components/card-footer';
import f7CardHeader from './components/card-header';
import f7Card from './components/card';
import f7Checkbox from './components/checkbox';
import f7Chip from './components/chip';
import f7Col from './components/col';
import f7FabBackdrop from './components/fab-backdrop';
import f7FabButton from './components/fab-button';
import f7FabButtons from './components/fab-buttons';
import f7Fab from './components/fab';
import f7Gauge from './components/gauge';
import f7Icon from './components/icon';
import f7Input from './components/input';
import f7Link from './components/link';
import f7ListButton from './components/list-button';
import f7ListGroup from './components/list-group';
import f7ListIndex from './components/list-index';
import f7ListInput from './components/list-input';
import f7ListItemCell from './components/list-item-cell';
import f7ListItemRow from './components/list-item-row';
import f7ListItem from './components/list-item';
import f7List from './components/list';
import f7LoginScreenTitle from './components/login-screen-title';
import f7LoginScreen from './components/login-screen';
import f7MenuDropdownItem from './components/menu-dropdown-item';
import f7MenuDropdown from './components/menu-dropdown';
import f7MenuItem from './components/menu-item';
import f7Menu from './components/menu';
import f7Message from './components/message';
import f7MessagebarAttachment from './components/messagebar-attachment';
import f7MessagebarAttachments from './components/messagebar-attachments';
import f7MessagebarSheetImage from './components/messagebar-sheet-image';
import f7MessagebarSheetItem from './components/messagebar-sheet-item';
import f7MessagebarSheet from './components/messagebar-sheet';
import f7Messagebar from './components/messagebar';
import f7MessagesTitle from './components/messages-title';
import f7Messages from './components/messages';
import f7NavLeft from './components/nav-left';
import f7NavRight from './components/nav-right';
import f7NavTitleLarge from './components/nav-title-large';
import f7NavTitle from './components/nav-title';
import f7Navbar from './components/navbar';
import f7PageContent from './components/page-content';
import f7Page from './components/page';
import f7Panel from './components/panel';
import f7PhotoBrowser from './components/photo-browser';
import f7PieChart from './components/pie-chart';
import f7Popover from './components/popover';
import f7Popup from './components/popup';
import f7Preloader from './components/preloader';
import f7Progressbar from './components/progressbar';
import f7Radio from './components/radio';
import f7Range from './components/range';
import f7RoutableModals from './components/routable-modals';
import f7Row from './components/row';
import f7Searchbar from './components/searchbar';
import f7Segmented from './components/segmented';
import f7Sheet from './components/sheet';
import f7SkeletonAvatar from './components/skeleton-avatar';
import f7SkeletonBlock from './components/skeleton-block';
import f7SkeletonImage from './components/skeleton-image';
import f7SkeletonText from './components/skeleton-text';
import f7Stepper from './components/stepper';
import f7Subnavbar from './components/subnavbar';
import f7SwipeoutActions from './components/swipeout-actions';
import f7SwipeoutButton from './components/swipeout-button';
import f7SwiperSlide from './components/swiper-slide';
import f7Swiper from './components/swiper';
import f7Tab from './components/tab';
import f7Tabs from './components/tabs';
import f7TextEditor from './components/text-editor';
import f7Toggle from './components/toggle';
import f7Toolbar from './components/toolbar';
import f7TreeviewItem from './components/treeview-item';
import f7Treeview from './components/treeview';
import f7UseIcon from './components/use-icon';
import f7View from './components/view';
import f7Views from './components/views';

export { f7AccordionContent, f7AccordionItem, f7AccordionToggle, f7Accordion, f7ActionsButton, f7ActionsGroup, f7ActionsLabel, f7Actions, f7App, f7Appbar, f7AreaChart, f7Badge, f7BlockFooter, f7BlockHeader, f7BlockTitle, f7Block, f7Button, f7CardContent, f7CardFooter, f7CardHeader, f7Card, f7Checkbox, f7Chip, f7Col, f7FabBackdrop, f7FabButton, f7FabButtons, f7Fab, f7Gauge, f7Icon, f7Input, f7Link, f7ListButton, f7ListGroup, f7ListIndex, f7ListInput, f7ListItemCell, f7ListItemRow, f7ListItem, f7List, f7LoginScreenTitle, f7LoginScreen, f7MenuDropdownItem, f7MenuDropdown, f7MenuItem, f7Menu, f7Message, f7MessagebarAttachment, f7MessagebarAttachments, f7MessagebarSheetImage, f7MessagebarSheetItem, f7MessagebarSheet, f7Messagebar, f7MessagesTitle, f7Messages, f7NavLeft, f7NavRight, f7NavTitleLarge, f7NavTitle, f7Navbar, f7PageContent, f7Page, f7Panel, f7PhotoBrowser, f7PieChart, f7Popover, f7Popup, f7Preloader, f7Progressbar, f7Radio, f7Range, f7RoutableModals, f7Row, f7Searchbar, f7Segmented, f7Sheet, f7SkeletonAvatar, f7SkeletonBlock, f7SkeletonImage, f7SkeletonText, f7Stepper, f7Subnavbar, f7SwipeoutActions, f7SwipeoutButton, f7SwiperSlide, f7Swiper, f7Tab, f7Tabs, f7TextEditor, f7Toggle, f7Toolbar, f7TreeviewItem, f7Treeview, f7UseIcon, f7View, f7Views }
export { f7, f7ready, theme, useStore };
export default Framework7Vue;

function registerComponents(app) {
  app.component('f7-accordion-content', f7AccordionContent);
  app.component('f7-accordion-item', f7AccordionItem);
  app.component('f7-accordion-toggle', f7AccordionToggle);
  app.component('f7-accordion', f7Accordion);
  app.component('f7-actions-button', f7ActionsButton);
  app.component('f7-actions-group', f7ActionsGroup);
  app.component('f7-actions-label', f7ActionsLabel);
  app.component('f7-actions', f7Actions);
  app.component('f7-app', f7App);
  app.component('f7-appbar', f7Appbar);
  app.component('f7-area-chart', f7AreaChart);
  app.component('f7-badge', f7Badge);
  app.component('f7-block-footer', f7BlockFooter);
  app.component('f7-block-header', f7BlockHeader);
  app.component('f7-block-title', f7BlockTitle);
  app.component('f7-block', f7Block);
  app.component('f7-button', f7Button);
  app.component('f7-card-content', f7CardContent);
  app.component('f7-card-footer', f7CardFooter);
  app.component('f7-card-header', f7CardHeader);
  app.component('f7-card', f7Card);
  app.component('f7-checkbox', f7Checkbox);
  app.component('f7-chip', f7Chip);
  app.component('f7-col', f7Col);
  app.component('f7-fab-backdrop', f7FabBackdrop);
  app.component('f7-fab-button', f7FabButton);
  app.component('f7-fab-buttons', f7FabButtons);
  app.component('f7-fab', f7Fab);
  app.component('f7-gauge', f7Gauge);
  app.component('f7-icon', f7Icon);
  app.component('f7-input', f7Input);
  app.component('f7-link', f7Link);
  app.component('f7-list-button', f7ListButton);
  app.component('f7-list-group', f7ListGroup);
  app.component('f7-list-index', f7ListIndex);
  app.component('f7-list-input', f7ListInput);
  app.component('f7-list-item-cell', f7ListItemCell);
  app.component('f7-list-item-row', f7ListItemRow);
  app.component('f7-list-item', f7ListItem);
  app.component('f7-list', f7List);
  app.component('f7-login-screen-title', f7LoginScreenTitle);
  app.component('f7-login-screen', f7LoginScreen);
  app.component('f7-menu-dropdown-item', f7MenuDropdownItem);
  app.component('f7-menu-dropdown', f7MenuDropdown);
  app.component('f7-menu-item', f7MenuItem);
  app.component('f7-menu', f7Menu);
  app.component('f7-message', f7Message);
  app.component('f7-messagebar-attachment', f7MessagebarAttachment);
  app.component('f7-messagebar-attachments', f7MessagebarAttachments);
  app.component('f7-messagebar-sheet-image', f7MessagebarSheetImage);
  app.component('f7-messagebar-sheet-item', f7MessagebarSheetItem);
  app.component('f7-messagebar-sheet', f7MessagebarSheet);
  app.component('f7-messagebar', f7Messagebar);
  app.component('f7-messages-title', f7MessagesTitle);
  app.component('f7-messages', f7Messages);
  app.component('f7-nav-left', f7NavLeft);
  app.component('f7-nav-right', f7NavRight);
  app.component('f7-nav-title-large', f7NavTitleLarge);
  app.component('f7-nav-title', f7NavTitle);
  app.component('f7-navbar', f7Navbar);
  app.component('f7-page-content', f7PageContent);
  app.component('f7-page', f7Page);
  app.component('f7-panel', f7Panel);
  app.component('f7-photo-browser', f7PhotoBrowser);
  app.component('f7-pie-chart', f7PieChart);
  app.component('f7-popover', f7Popover);
  app.component('f7-popup', f7Popup);
  app.component('f7-preloader', f7Preloader);
  app.component('f7-progressbar', f7Progressbar);
  app.component('f7-radio', f7Radio);
  app.component('f7-range', f7Range);
  app.component('f7-routable-modals', f7RoutableModals);
  app.component('f7-row', f7Row);
  app.component('f7-searchbar', f7Searchbar);
  app.component('f7-segmented', f7Segmented);
  app.component('f7-sheet', f7Sheet);
  app.component('f7-skeleton-avatar', f7SkeletonAvatar);
  app.component('f7-skeleton-block', f7SkeletonBlock);
  app.component('f7-skeleton-image', f7SkeletonImage);
  app.component('f7-skeleton-text', f7SkeletonText);
  app.component('f7-stepper', f7Stepper);
  app.component('f7-subnavbar', f7Subnavbar);
  app.component('f7-swipeout-actions', f7SwipeoutActions);
  app.component('f7-swipeout-button', f7SwipeoutButton);
  app.component('f7-swiper-slide', f7SwiperSlide);
  app.component('f7-swiper', f7Swiper);
  app.component('f7-tab', f7Tab);
  app.component('f7-tabs', f7Tabs);
  app.component('f7-text-editor', f7TextEditor);
  app.component('f7-toggle', f7Toggle);
  app.component('f7-toolbar', f7Toolbar);
  app.component('f7-treeview-item', f7TreeviewItem);
  app.component('f7-treeview', f7Treeview);
  app.component('f7-use-icon', f7UseIcon);
  app.component('f7-view', f7View);
  app.component('f7-views', f7Views)
}
export { registerComponents }
